src/WebJobs.Script.Grpc/Channel/OrderedInvocationMessageDispatcher.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Grpc.Eventing;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc;
+
+internal sealed class OrderedInvocationMessageDispatcher : IInvocationMessageDispatcher, IDisposable
+{
+    private readonly ILogger _logger;
+    private readonly string _invocationId;
+    private readonly Action<InboundGrpcEvent> _dispatchWithChannel;
+
+    // Separated these out for easier testing of the flow.
+    private readonly Action<InboundGrpcEvent> _dispatchWithThreadPool;
+
+    private Channel<InboundGrpcEvent> _channel;
+    private bool _isChannelInitialized = false;
+    private bool _invocationComplete = false;
+    private bool _disposed = false;
+
+    public OrderedInvocationMessageDispatcher(string invocationId, ILogger logger, Action<InboundGrpcEvent> processItem)
+        : this(invocationId, logger, processItem, processItem)
+    {
+    }
+
+    // For testing
+    internal OrderedInvocationMessageDispatcher(string invocationId, ILogger logger, Action<InboundGrpcEvent> processItemWithChannel,
```

Comment:
Can we add summary comments to all the ctors & class? And especially for the test items i.e.

```
/// <summary>
/// Initializes a new instance of the <see cref="OrderedInvocationMessageDispatcher"/> class for mocking use in testing.
/// </summary>
/// <remarks>
/// This constructor exists only to support mocking. When used, class state is not fully initialized, and
/// will not function correctly; virtual members are meant to be mocked.
///</remarks>
```